tests/functional/conftest.py

```diff
@@ -343,27 +345,59 @@ def _run_executable(self, prog, args, run_from_path, runtime):
         # Windows command prompt. Py.test is then able to collect stdout/sterr
         # messages and display them if a test fails.
 
-        process = psutil.Popen(args, executable=exe_path, stdout=sys.stdout,
-                               stderr=sys.stderr, env=prog_env, cwd=prog_cwd)
-        # 'psutil' allows to use timeout in waiting for a subprocess.
-        # If not timeout was specified then it is 'None' - no timeout, just waiting.
-        # Runtime is useful mostly for interactive tests.
-        try:
-            timeout = runtime if runtime else _EXE_TIMEOUT
-            retcode = process.wait(timeout=timeout)
-        except psutil.TimeoutExpired:
+        # Note that we cannot let Popen get ahold of sys.stderr/out because it
+        # will attempt to close the files with nonstandard (system-level) 
+        # methods via fileno(). This causes problems on win32 from which pytest
+        # cannot correctly recover. 
```

Comment:
Correct. Even passing in my own tempfiles did not fix the problem on py3 because they were closed early.